### PR TITLE
Fix #198: Address javadoc warnings from the codegen module

### LINKED ISSUE #198: Clean up the javadoc warnings
There are screeds of javadoc warnings when running the maven build. 

* This has a negative impact on productivity. 
* Developers are less likely to notice impactful warnings amongst the noise. 
* Our javadoc will tend to rot, as there'll be an assumption that we don't care about it.  

We should clean up the warnings.

Run `mvn verify` and look for the screeds of yellow.

_Originally posted by @k-wall in https://github.com/kroxylicious/kroxylicious/issues/195#issuecomment-1467944993_

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

Contribute javadoc for the codegen module.  No functional change.

* backfill missing javadoc
* in some cases, reducing visibility of implementation seemed like the best course
* added package level note explaining why copies of classes from org.apache.kafka.message appear in io.kroxylicious.krpccodegen.schema. I purposefully didn't add javadoc to these classes. Instead, I excluded the package from javadoc'ing.

The module could have utility for filter author wishing to autogenerate filter code.  This is an additional reason to value the javadoc of this model.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [x] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: krpc-code-gen/src/main/java/io/kroxylicious/krpccodegen/main/KrpcGenerator.java [hunk 1]
```diff
@@ -42,8 +42,14 @@
 import freemarker.template.TemplateExceptionHandler;
 import freemarker.template.Version;
 
+/**
+ * Code generator driven by Apache Kafka message specifications definitions.
+ */
 public class KrpcGenerator {
 
+    /**
+     * Builder for the {@link KrpcGenerator}.
```
Comment (SamBarker) [inline krpc-code-gen/src/main/java/io/kroxylicious/krpccodegen/main/KrpcGenerator.java:51]:
```suggestion
     * Configures and instantiates the {@link KrpcGenerator}.
```

?

**Review by SamBarker — APPROVED** [review]
LGTM

**Review by robobario — APPROVED** [review]
LGTM

**Comment by robobario** [comment]
I had a look at the support for unit testing Mojos to get up the coverage and it is not great. There is an old test harness last released 8 years ago that sort of worked when I added a maven- compat jar (I assume it predates maven3 and is using deprecated APIs that need this compat lib) it also uses junit 4 classes. Generally ancient and to be avoided.

**Comment by k-wall** [comment]
I ignored the code coverage complaint from Sonar.  I don't want to address the lack of coverage for the maven plugin now.

**Pull request merged**